### ISSUE TITLE
Make plugin compatible with Grails 2.4.X

### ISSUE THREAD

**Issue by kkorsakov** [goeh/grails-crm-core#1]
Hi!

I know that all samples are written for Grails 2.2.4 and i suppose there were some reasons to do so.
But i want to run CRM with Grails 2.4.4 (latest stable version for current time) because i already have a  project based on this version:)

I don't think this task is very hard, but i first would like to know what are the reasons, did you try to upgrade version, do you think it will be a good improvement for CRM?

Thank you!

**Comment by goeh** [comment]
Trust me, I also want to use Grails 2.4.4 with GR8 CRM. I have tried to upgrade to Grails 2.4.x several times but failed due to incompatible external plugins. Initially I got stuck on the shiro plugin, but I think the latest release should work with Grails 2.4.4. Last time I tried with Grails 2.4.4 I could not get it to compile with cache-ehcache and hibernate4, if I remember correctly.
I will give it some time this week and see if I can make any progress.

**Comment by goeh** [comment]
I started to fix plugins last night. I had issues with the following plugins:
- ehcache
- shiro
- navigation

Step by step I fixed or worked around them. Some are fixed by others on github but not released yet. Now I'm struggling with issues on:
- resources
- less-resources

I will continue working through the problems this week.

**Comment by kkorsakov** [comment]
I'm working on it locally also. Finally it starts and working (as much as i can test it). 
My strategy was to exclude all non_working/already_unsupported plugins and override them with newer versions or exclude at all. 
This is my list of problems/solutions, maybe it can help you:
- static compilation errors with Groovy 2.3 bundled with Grails 2.4 in several classes of crm-core
- hibernate doesn't work with 2.4; must be excluded in favor of hibernate4
- cache must be upgraded to cache:1.1.8
- navigation uses legacy Holders - must be excluded (application must be rewritten to use platform-core?)
- jquery and fields must be upgraded to latest versions
- resources and less-resources. I think that for a long-term perspective asset pipeline (and its less compiler, etc) should be used, how do you think? I know there are problems with resources, less and bootstrap 3.x. 
  Actually I'm going this way but there are lots of changes in gsp's. This is the largest part of upgrade i suppose. 
  I didn't have any problems with shiro. Version is 1.2.1.

Maybe we can coordinate our efforts somehow now or in the future? I'm interested in using CRM in at least two own projects. At now i modify CRM code locally and plug it using grails inline plugin override mechanism, but it may be more productive to contribute changes.

**Comment by ivgsilva** [comment]
Hello I am following the discussion because I'm also in a situation like the kkorsakov.

Would be good if some of this update to version 2.4.

Use the 2.3.7 version and it is not compatible.

Goeh success.

**Comment by goeh** [comment]
Sorry for being quiet. I've been busy with work but I did find some time to migrate GR8 CRM plugins to Grails 2.4.4. Most plugins are now migrated and initial tests with Grails 2.4.4 indicate success. See http://gr8crm.github.io for a list of upgraded plugins.

The process was stalled for several days waiting for my pull request on navigation plugin to be accepted. But I got no response so I migrated all navigation stuff to the navigation support in platform-core plugin. GR8 CRM uses platform-core for application events so it was a natural move.

The demo application https://github.com/technipelago/gr8crm-demo-app is upgraded to Grails 2.4.4
